src/ps/psreaddata.cpp

```diff
@@ -588,59 +627,105 @@ PetscErrorCode PSReadMatPowerData(PS ps, const char netfile[]) {
 
     if (bus_start_line != -1 && bus_end_line == -1) {
       if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
-        bus_nblank_lines++;
+        bus_nblank_lines = bus_nblank_lines + 1;
     }
     if (gen_start_line != -1 && gen_end_line == -1) {
       if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
-        gen_nblank_lines++;
+        gen_nblank_lines = gen_nblank_lines + 1;
     }
 
     if (br_start_line != -1 && br_end_line == -1) {
       if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
-        br_nblank_lines++;
+        br_nblank_lines = br_nblank_lines + 1;
+    }
+
+    if (dcline_start_line != -1 && dcline_end_line == -1) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        dcline_nblank_lines = dcline_nblank_lines + 1;
+    }
+
+    if (gencost_start_line != -1 && gencost_end_line == -1) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        gencost_nblank_lines = gencost_nblank_lines + 1;
+    }
+
+    if (loadcost_start_line != -1 && loadcost_end_line == -1) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        loadcost_nblank_lines = loadcost_nblank_lines + 1;
+    }
+
+    if (genfuel_start_line != -1 && genfuel_end_line == -1) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        genfuel_nblank_lines = genfuel_nblank_lines + 1;
     }
 
     /* Count the number of pq loads */
     if (bus_start_line != -1 && line_counter >= bus_start_line &&
         bus_end_line == -1) {
       sscanf(line, "%d %d %lf %lf", &extbusnum, &bustype_i, &Pd, &Qd);
+      if (bustype_i == ISOLATED_BUS) { /* Isolated bus */
+        /* put it in the isolated_buses array for later work */
+        ps->isolated_buses[ps->nisolated_buses++] = extbusnum;
+        if (ps->nisolated_buses == buff_num * 10) {
+          /* Reached buffer max, need to allocate more size */
+          buff_num++;
+          ierr = PetscCalloc1(buff_num * 10, &temp_buff);
+          CHKERRQ(ierr);
+          /* Copy over isolated_buses to temp_buff */
+          ierr = PetscMemcpy(temp_buff, ps->isolated_buses,
+                             (buff_num - 1) * 10 * sizeof(PetscInt));
+          ierr = PetscFree(ps->isolated_buses);
+          CHKERRQ(ierr);
+          /* Updated pointer to isolated_buses */
+          ps->isolated_buses = temp_buff;
+        }
+      }
+
       if (!((Pd == 0.0) && (Qd == 0.0)))
-        ps->Nload++;
+        load_numlines++;
```

Comment:
Similarly here, this is incremented, but not used elsewhere?

Reply:
It is used here `ierr = PetscCalloc1(load_numlines, &ps->load);`